src/Compilers/CSharp/Portable/BoundTree/BoundNode_Source.cs

```diff
@@ -0,0 +1,419 @@
+                                    appendSource(statement);
+                                }
+
+                                appendLine("break;");
+
+                                decrementIndent();
+                            }
+                            appendLine("}");
+                            break;
+                        }
+                    case BoundSwitchLabel label:
+                        {
+                            appendSource(label.ExpressionOpt);
+                            break;
+                        }
+                    case BoundUnaryOperator unary:
+                        {
+                            append($" {unary.OperatorKind.ToString()} ");
+                            appendSource(unary.Operand);
+                            break;
+                        }
+                    case BoundStatementList list:
+                        {
+                            foreach (var statement in list.Statements)
+                            {
+                                appendSource(statement);
+                            }
+                            break;
+                        }
+                    case BoundSequence sequence:
+                        {
+                            append("{ ");
+                            foreach (var effect in sequence.SideEffects)
+                            {
+                                appendSource(effect);
+                                append("; ");
+                            }
+                            appendSource(sequence.Value);
+                            append(" }");
+                            break;
+                        }
+                    case BoundDefaultExpression _:
+                        {
+                            append("default");
+                            break;
+                        }
+                    case BoundBinaryOperator binary:
+                        {
+                            appendSource(binary.Left);
+                            append(" ");
+                            append(binary.OperatorKind.ToString());
+                            append(" ");
+                            appendSource(binary.Right);
+                            break;
+                        }
+                    default:
+                        appendLine(node.Kind.ToString());
+                        break;
+                }
+
+                void appendSource(BoundNode n)
+                {
+                    if (n is null)
+                    {
+                        append("NULL");
+                    }
+                    else
+                    {
+                        appendSourceCore(n, indent, tempIdentifiers);
+                    }
+                }
+
+                void append(string s)
+                {
+                    builder.Append(s);
+                }
+
+                void incrementIndent()
+                {
+                    indent += 4;
```

Comment:
Instead of hard coding 4 everywhere consider making indent size a constant for the outer function.